File: 프로그래머스/0/120883. 로그인 성공？/로그인 성공？.java
class Solution {
    public String solution(String[] id_pw, String[][] db) {
        boolean flag1 = false;
        boolean flag2 = false;
        String ret = "";
        for(String[] srr : db){
            if(id_pw[0].equals(srr[0]) && id_pw[1].equals(srr[1])) return "login";
            else if(id_pw[0].equals(srr[0]) && !id_pw[1].equals(srr[1])) ret = "wrong pw";
            else if(!id_pw[0].equals(srr[0]) && !ret.equals("wrong pw")) ret = "fail";
        }
        return ret;
    }
}
/*
로그인 성공 -> 아이디 비밀번호 일치 -> login
로그인 실패 -> 아이디 일치x -> fail
         -> 아이디 일치O && 비밀번호 x -> wrong pw;
*/

File: 프로그래머스/0/120883. 로그인 성공？/로그인 성공？.py
def solution(id_pw, db):
    db = {x[0] : x[1] for x in db}
    if id_pw[0] not in db.keys():
        return "fail"
    elif db[id_pw[0]] != id_pw[1]:
        return "wrong pw"
    else:
        return "login"
    
    
"""
1. 아이디 비밀번호 일치는 로그인
2. 로그인 실패
    2-1. 아이디 불일치: fail
    2-2. 아이디 일치 & 비밀번호 불일치: wron pw 
"""